pulp_rpm/app/models.py

```diff
@@ -678,3 +678,43 @@ class RpmDistribution(PublicationDistribution):
     """
 
     TYPE = 'rpm'
+
+
+class Modulemd(Content):
+    """
+    The "Modulemd" content type. Modularity support.
+
+    Fields:
+        name (Text):
+            Name of the modulemd
+        stream (Text):
+            The modulemd's stream
+        version (Text):
+            The version of the modulemd.
+        context (Text):
+            The context flag serves to distinguish module builds with the
+            same name, stream and version and plays an important role in
+            future automatic module stream name expansion.
+        arch (Text):
+            Contains a string describing the module's artifacts' main hardware
+            architecture compatibility.
+        dependencies (Text):
+            Module dependencies, if any.
+        artifacts (List):
+            Artifacts shipped with this module.
+        packages (List):
+            List of RPM Packages connected to this modulemd.
+    """
+
+    TYPE = "modulemd"
+
+    # required metadata
+    name = models.CharField(max_length=255)
+    stream = models.CharField(max_length=255)
+    version = models.CharField(max_length=255)
+    context = models.CharField(max_length=255)
+    arch = models.CharField(max_length=255)
+
+    dependencies = models.TextField(default='[]')
+    artifacts = models.TextField(default='[]')
+    packages = models.ManyToManyField(Package)
```

Comment:
Can an RPM be part of more than one module?  It would be great if we can use a one-to-many relationship instead of a many-to-many relationship.

Reply:
Also let's prefix it _packages since it's not part of the actual modulemd metadata.

Reply:
Do we have a specific use case for storing this relationship in the DB, such as manual copy?  I think with libsolv the flat textual list of artifacts provides sufficient information that this isn't necessary, but I'm not sure if another use case requires it.

Reply:
packages changed to have '_'.
We want to keep packages (as what we have in pulp) for copy use case. 
Depend how we implement 'copy' later on.

Reply:
@dralley, use cases off the top of my head: removal of modulemd, simple copy, applicability.